Readme.md

```diff
@@ -1,21 +1,146 @@
-# Aragon OSx
+![Aragon](https://res.cloudinary.com/dacofvu8m/image/upload/v1677353961/Aragon%20CodeArena/osx_blue_logo_lqrvkr.png)
 
-This workspace contains the foundation of the Aragon OSx ecosystem - the human centered approach to DAO's.
+# Aragon OSx Protocsol
 
-For more information on the individual packages, please read the respective `Readme.md`.
+The Aragon OSx protocol is the foundation layer of the new Aragon stack. It allows users to create, manage, and customize DAOs in a way that is lean, adaptable, and secure.
+
+Within this monorepo, you will be able to find 3 individual packages:
+- [Contracts](https://github.com/aragon/osx/tree/develop/packages/contracts): the Aragon OSx protocol contracts.
+- [Subgraph](https://github.com/aragon/osx/tree/develop/packages/subgraph): contains all code generating our subgraph and event indexing.
+- [Contract-ethers](https://github.com/aragon/osx/tree/develop/packages/contracts-ethers): contains the connection between the ethers package and our contracts.
+
+For more information on the individual packages, please read the respective `README.md`.
+
+## Contributing
+
+We'd love to hear what you think! If you want to build this with us, please find a detailed contribution guide in the `CONTRIBUTION_GUIDE.md` [file here](https://github.com/aragon/osx/blob/develop/CONTRIBUTION_GUIDE.md).
 
 ## Setup
 
-Start by running `yarn install` in the root.
+Start by running `yarn install` in the project root in your terminal.
 
 ### Dependencies
 
 Since the repo is set up as yarn workspace, all the linking is done automatically.
 
+## How the Aragon OSx protocol works
+
+To review the contracts powering the Aragon OSx protocol, feel free to head to `packages/contracts`.
+
+The Aragon OSx protocol architecture is composed of two key sections:
+
+- __Core contracts__: the primitives the end user will interact with. It is composed of mostly 3 sections:
+    - **DAO contract:** the main contract of our core. It holds a DAO's assets and possible actions.
+    - **Permissions**: govern interactions between the plugins, DAOs, and any other address - allowing them (or not) to execute actions on behalf of and within the DAO.
+    - **Plugins**: base templates of plugins to build upon.
+- __Framework contracts__: in charge of creating and registering each deployed DAO or plugin. It contains:
+    - **DAO and Plugin Factories**: creates DAOs or plugins.
+    - **DAO and Plugin Registries**: registers into our protocol those DAOs or plugins.
+    - **Plugin Processor:** installs and uninstalls plugins into DAOs.
```

Comment:
```suggestion
    - **Plugin Setup Processor:** installs and uninstalls plugins into DAOs.
```

It processes the plugin setups that preprare the plugin for installation/update/deletion